### PR TITLE
feat: Added support for Authenticated Datafiles

### PR DESCRIPTION
**Pull request by zashraf1985**
## Summary
Added support for Authenticated Datafiles

## Test plan
Added new unit tests

### CODE REVIEW

**Comment by zashraf1985** [comment]
Hey @thomaszurkan-optimizely . Thanks for reviewing quickly. I am already writing tests. That's what I mentioned in the PR summary too.

File: spec/config_manager/http_project_config_manager_spec.rb [hunk 1]
```diff
@@ -468,4 +463,56 @@
...
+
+    it 'should add authorization header when auth token is provided' do
+      allow(Optimizely::Helpers::HttpUtils).to receive(:make_request)
+      @http_project_config_manager = Optimizely::HTTPProjectConfigManager.new(
+        sdk_key: 'valid_sdk_key',
+        datafile_access_token: 'the-token'
+      )
+      sleep 0.1
+      expect(Optimizely::Helpers::HttpUtils).to have_received(:make_request).with(anything, anything, anything, hash_including('Authorization' => 'Bearer the-token'), anything)
+    end
```
Comment (pthompson127) [inline spec/config_manager/http_project_config_manager_spec.rb:486]: Just out of curiosity, is there a reason why this part of the spec is duplicated? Seems like the two blocks are exactly the same. 
Wondering if this is instead supposed to test the case where an invalid SDK key is provided (not sure if that is handled / error-checked elsewhere)

Comment (zashraf1985) [reply spec/config_manager/http_project_config_manager_spec.rb:486]: Hey @pthompson127 ... good eye... its just a plain stupid mistake :). i wil fix it.

Comment (pthompson127) [reply spec/config_manager/http_project_config_manager_spec.rb:486]: @zashraf1985 ahh gotcha :)

**Pull request merged**